GitUpKit/Views/GIAdvancedCommitViewController.m

```diff
@@ -187,16 +187,24 @@ - (void)diffFilesViewControllerDidChangeSelection:(GIDiffFilesViewController*)co
 
 - (void)_stageSelectedFiles:(NSArray*)selectedDeltas {
   NSMutableArray* deltas = [[NSMutableArray alloc] init];
+  NSMutableArray* nonSubmoduleDeltasPaths = [[NSMutableArray alloc] init];
+  NSMutableArray* nonSubmoduleDeltas = [[NSMutableArray alloc] init];
+
   for (GCDiffDelta* delta in selectedDeltas) {
     if (![_indexConflicts objectForKey:delta.canonicalPath]) {
       if (delta.submodule) {
         [self stageSubmoduleAtPath:delta.canonicalPath];
+        [deltas addObject:delta];
       } else {
-        [self stageAllChangesForFile:delta.canonicalPath];
+        [nonSubmoduleDeltasPaths addObject:delta.canonicalPath];
+        [nonSubmoduleDeltas addObject:delta];
       }
-      [deltas addObject:delta];
     }
```

Comment:
Seems like we can just have 2 arrays, `deltas` and `nonSubmoduleDeltasPaths`. In the end we're adding all the `deltas` together but we keep track of `nonSubmoduleDeltasPaths` to stage them all in bulk. I don't see why we need `nonSubmoduleDeltas`.

Reply:
Ah yes that totally makes sense